backend/server.go

```diff
@@ -0,0 +1,183 @@
+package backend
+
+import (
+	rice "github.com/GeertJohan/go.rice"
+	"github.com/gomodule/redigo/redis"
+	"html/template"
+	"log"
+	"net/http"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+type Data struct {
+	People []Person
+	Awards []string
+}
+
+type Person struct {
+	Name string `redis:"name"`
+	Score float64 `redis:"score"`
+}
+
+type App struct {
+	pool *redis.Pool
+	config Config
+	tmpl 	template.Template
+}
+
+func renderTemplate(w http.ResponseWriter, people []Person, tmpl template.Template) {
+	err := tmpl.ExecuteTemplate(w, "index.html", Data{
+		People: people,
+		Awards: awards,
+	})
+	checkError(w, err)
+}
+
+func checkError(w http.ResponseWriter, err error) {
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+}
+
+func (app *App) homeHandler(w http.ResponseWriter, r *http.Request) {
+	people, err := app.loadPeople()
+	checkError(w, err)
+
+	renderTemplate(w, people, app.tmpl)
+
+	awards = make([]string ,0)
+}
+
+func (app *App) addHandler(w http.ResponseWriter, r *http.Request) {
+	conn := app.pool.Get()
+
+	err := r.ParseForm()
+	checkError(w, err)
+
+	for key, values := range r.Form {
+		name := strings.SplitN(key, "-", 2)[1]
+		value := values[0]
+
+		if value == "" {
+			continue
+		}
+
+		byPoints, err := strconv.ParseFloat(value, 64)
+		checkError(w, err)
+
+		// check awards
+		currScore, err := redis.Float64(conn.Do("HGET", name, "score"))
+		checkError(w, err)
+		if currScore + byPoints >= 100 {
+			awards = append(awards, name)
+			byPoints -= 100
+		}
+		_, err = conn.Do("HINCRBYFLOAT", name, "score", byPoints)
+		checkError(w, err)
+	}
+	http.Redirect(w, r, "/", http.StatusFound)
+}
+
+func (app *App) addPeopleHandler(w http.ResponseWriter, r *http.Request){
+	conn := app.pool.Get()
+
+	name, score, expression := r.FormValue("new-name"), r.FormValue("new-score"), r.FormValue("new-score-exp")
+
+	existed, err := redis.Bool(conn.Do("EXISTS", name))
+	checkError(w, err)
+
+	if !existed {
+		// check award
+		scoreVal, err := strconv.ParseFloat(score, 64)
+		if scoreVal >= 100 {
+			awards = append(awards, name)
+			scoreVal -= 100
```

Comment:
Some of the ways the System⚙️🔮 has functioned previously:
* Everyone gets reset to 0 on Quacktastic win
* Everyone keeps their existing points but the winner goes to 0
* Some (Efe, Uttara) have permanent negative or complex points. 

A customizable global reset which is triggered when someone crosses 100 might be a fun idea.